dev/com.ibm.ws.microprofile.4.0_fat/fat/src/com/ibm/ws/microprofile40/fat/suite/MPCompatibleTest.java

```diff
@@ -0,0 +1,295 @@
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile40.fat.suite;
+
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.PropertiesAsset;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.ws.microprofile40.fat.tests.helloworld.HelloWorldApplication;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+@RunWith(FATRunner.class)
+public class MPCompatibleTest {
+
+    private static final String SERVER_NAME = "MPServer";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    private static final String APP_NAME = "helloworld";
+
+    private static final String MESSAGE = "Hello World!";
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        PropertiesAsset config = new PropertiesAsset().addProperty("message", MESSAGE);
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                                   .addPackage(HelloWorldApplication.class.getPackage())
+                                   .addAsResource(config, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, DeployOptions.SERVER_ONLY);
+    }
+
+    /**
+     * Just microProfile-4.0 ... Should always pass, a sanity check
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.LITE)
+    public void testMicroProfile40() throws Exception {
+        try {
+            server.setServerConfigurationFromFilePath("MP40.xml");
+            server.startServer();
+            runGetMethod(200, "/helloworld/helloworld", MESSAGE);
+        } finally {
+            server.stopServer("CWMOT0010W"); //CWMOT0010W: OpenTracing cannot track JAX-RS requests because an OpentracingTracerFactory class was not provided or client libraries for tracing backend are not in the class path.
```

Comment:
I am wondering whether a better fix is to add the jaeger client jar and add the classloader element in the server.xml so that this warning message will not appear any more.

Reply:
I think a better fix is to not output the message at all. I haven't configured a jaeger server so why would I need a jaeger client?

Reply:
I think the warning is about the partial open tracing client set up. With the client, the app will be able to correlate the trace span. It does not matter whether there is a server or not. It is like you have all info no matter you want to look at them or not. The client should not rely on server at all.